src/NUnitFramework/framework/Constraints/NUnitComparer.cs

```diff
@@ -58,23 +59,29 @@ public int Compare(object x, object y)
             if (Numerics.IsNumericType(x) && Numerics.IsNumericType(y))
                 return Numerics.Compare(x, y);
 
-            if (x is IComparable)
-                return ((IComparable)x).CompareTo(y);
-
-            if (y is IComparable)
-                return -((IComparable)y).CompareTo(x);
-
             Type xType = x.GetType();
             Type yType = y.GetType();
 
+            // If we use BindingFlags.ExactBinding it will prevent us finding CompareTo(object)
+            // It however also prevents finding CompareTo(TBase) when called with TDerived
+            // Nor will it find CompareTo(int) when called with a short.
+            // We fallback to explicitly exclude CompareTo(object)
+            bool IsIComparable(MethodInfo method) => method.GetParameters()[0].ParameterType == typeof(object);
+
             MethodInfo method = xType.GetMethod("CompareTo", new Type[] { yType });
-            if (method != null)
+            if (method != null && !IsIComparable(method))
                 return (int)method.Invoke(x, new object[] { y });
 
             method = yType.GetMethod("CompareTo", new Type[] { xType });
-            if (method != null)
+            if (method != null && !IsIComparable(method))
                 return -(int)method.Invoke(y, new object[] { x });
```

Comment:
I'm not sure this prefers `IComparable<object>` over `IComparable`, especially when explicitly implemented, but we might not care about that.